applications/CompressiblePotentialFlowApplication/tests/cpp_tests/test_compressible_potential_element.cpp

```diff
@@ -118,6 +148,70 @@ namespace Kratos {
       }
     }
 
+    KRATOS_TEST_CASE_IN_SUITE(EmbeddedCompressiblePotentialFlowElementCalculateLocalSystemRHS, CompressiblePotentialApplicationFastSuite)
+    {
+      Model this_model;
+      ModelPart& model_part = this_model.CreateModelPart("Main", 3);
+
+      GenerateCompressibleEmbeddedElement(model_part);
+      Element::Pointer pElement = model_part.pGetElement(1);
+
+      // Define the nodal values
+      std::array<double,3> potential({1.0, 2.0, 3.0});
+      // Define the distance values
+      std::array<double,3> level_set({1.0, -1.0, -1.0});
+
+      for (unsigned int i = 0; i < 3; i++){
+        pElement->GetGeometry()[i].FastGetSolutionStepValue(VELOCITY_POTENTIAL) = potential[i];
+        pElement->GetGeometry()[i].FastGetSolutionStepValue(GEOMETRY_DISTANCE) = level_set[i];
+      }
+
+      // Compute RHS and LHS
+      Vector RHS = ZeroVector(3);
+      Matrix LHS = ZeroMatrix(3, 3);
+
+      pElement->CalculateLocalSystem(LHS, RHS, model_part.GetProcessInfo());
+
+      std::vector<double> reference({0.125625, 0.0, -0.125625});
+
+      KRATOS_CHECK_VECTOR_NEAR(RHS, reference, 1e-6);
+    }
+
+    KRATOS_TEST_CASE_IN_SUITE(EmbeddedCompressiblePotentialFlowElementCalculateLocalSystemLHS, CompressiblePotentialApplicationFastSuite)
+    {
+      Model this_model;
+      ModelPart& model_part = this_model.CreateModelPart("Main", 3);
+
+      GenerateCompressibleEmbeddedElement(model_part);
+      Element::Pointer pElement = model_part.pGetElement(1);
+
+      // Define the nodal values
+      std::array<double,3> potential({1.0, 2.0, 3.0});
+      // Define the distance values
+      std::array<double,3> level_set({1.0, -1.0, -1.0});
+      for (unsigned int i = 0; i < 3; i++){
+        pElement->GetGeometry()[i].FastGetSolutionStepValue(VELOCITY_POTENTIAL) = potential[i];
+        pElement->GetGeometry()[i].FastGetSolutionStepValue(GEOMETRY_DISTANCE) = level_set[i];
+      }
+
+      // Compute RHS and LHS
+      Vector RHS = ZeroVector(3);
+      Matrix LHS = ZeroMatrix(3, 3);
+
+      pElement->CalculateLocalSystem(LHS, RHS, model_part.GetProcessInfo());
+
+      std::array<double, 9> reference_array({0.251249, -0.25125, 1.08455e-06, -0.25125, 0.502499, -0.25125, 1.08455e-06, -0.25125, 0.251249});
+      // Copying to a 3x3 matrix to check against LHS
+      Matrix reference(3,3);
```

Comment:
After speaking with @loumalouomega , seems like it is not possible to do so for the Matrix type.

Also he recommended to use double `{{ }}` to initialize std::arrays, as some compilers might give a warning.

I will open a PR cleaning up the tests to apply this and also to use the new macros.

Reply:
ok
@marcnunezc I recommend to use the double `{{ }}` sparsely, this is a BUG in older versions of CLANG
Eventually the compilers will be updated and then this will no longer be possible (but confuse developers)